kpi/models/asset_file.py

```diff
@@ -17,18 +21,32 @@ def upload_to(self, filename):
     more information, see
     https://docs.djangoproject.com/en/1.8/topics/migrations/#serializing-values
     """
-    return posixpath.join(
-        self.asset.owner.username, 'asset_files', self.asset.uid,
-        self.file_type, filename
-    )
+    return AssetFile.get_path(self.asset, self.file_type, filename)
+
 
+class AssetFile(OpenRosaManifestInterface, models.Model):
 
-class AssetFile(models.Model):
     # More to come!
     MAP_LAYER = 'map_layer'
+    FORM_MEDIA = 'form_media'
+
     TYPE_CHOICES = (
         (MAP_LAYER, MAP_LAYER),
+        (FORM_MEDIA, FORM_MEDIA),
     )
+
+    ALLOWED_MIME_TYPES = {
+        FORM_MEDIA: ('image', 'audio', 'video', 'text/csv', 'application/xml'),
+        MAP_LAYER: (
+            'text/csv',
+            'application/vnd.google-earth.kml+xml',
+            'application/vnd.google-earth.kmz',
+            'application/wkt',
+            'application/geojson',
```

Comment:
let's use the standard `application/geo+json` (see my other comment on the settings file where you add the MIME types)

Reply:
Done!